src/basic/virt.c

```diff
@@ -472,6 +472,20 @@ int detect_container(void) {
                 goto finish;
         }
 
+        /* Docker creates /.dockerenv or /run/.dockerenv inside the container.
+         * https://github.com/moby/moby/issues/18355 */
+        if (access("/.dockerenv", F_OK) >= 0 || access("/run/.dockerenv", F_OK) >= 0) {
+                r = VIRTUALIZATION_DOCKER;
+                goto finish;
+        }
+
+        /* Podman instead creates /run/.containerenv inside the container.
+         * https://github.com/containers/podman/issues/3586#issuecomment-661918679 */
+        if (access("/run/.containerenv", F_OK) >= 0) {
+                r = VIRTUALIZATION_PODMAN;
+                goto finish;
+        }
+
         /* "Official" way of detecting WSL https://github.com/Microsoft/WSL/issues/423#issuecomment-221627364 */
```

Comment:
i'd feel more comfortable with this if this was moved to some table, i.e. like the vm_table[] or the dmi_vendor_table[], i.e. go through a loop of files to access(F_OK) on